internal/provider/instance.go

```diff
@@ -83,6 +84,11 @@ func (i Instance) FromMachine(m maas.Machine) Instance {
 	return i
 }
 
+// GetMetadata implements the Endpoint interface
+func (i *Instance) GetMetadata() interface{} {
+	return maas.Machine{}
```

Comment:
I feel like you want to return `maas.Machine` here

Reply:
I think you're right.